plugins/database/hana/hana.go

```diff
@@ -3,38 +3,37 @@ package hana
 import (
 	"context"
 	"database/sql"
-	"errors"
 	"fmt"
 	"strings"
-	"time"
 
-	_ "github.com/SAP/go-hdb/driver"
 	"github.com/hashicorp/vault/api"
-	"github.com/hashicorp/vault/sdk/database/dbplugin"
 	"github.com/hashicorp/vault/sdk/database/helper/connutil"
 	"github.com/hashicorp/vault/sdk/database/helper/credsutil"
 	"github.com/hashicorp/vault/sdk/database/helper/dbutil"
+	"github.com/hashicorp/vault/sdk/database/newdbplugin"
 	"github.com/hashicorp/vault/sdk/helper/dbtxn"
 	"github.com/hashicorp/vault/sdk/helper/strutil"
+
+	_ "github.com/SAP/go-hdb/driver"
 )
 
 const (
-	hanaTypeName = "hdb"
+	hanaTypeName        = "hdb"
+	maxIdentifierLength = 127
 )
 
 // HANA is an implementation of Database interface
 type HANA struct {
 	*connutil.SQLConnectionProducer
-	credsutil.CredentialsProducer
 }
 
-var _ dbplugin.Database = &HANA{}
+var _ newdbplugin.Database = &HANA{}
```

Comment:
This is now getting changed slightly in the other db plugins, so there's no unnecessary allocation:
```suggestion
var _ newdbplugin.Database = (*HANA)(nil)
```